core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2872,6 +2872,8 @@ private File getRootDirFor(String name) {
      * @throws AccessDeniedException as per {@link ItemGroup#getItem}
      */
     public @CheckForNull <T extends Item> T getItemByFullName(String fullName, Class<T> type) throws AccessDeniedException {
+        if (fullName == null) return null;
```

Comment:
Why would we want this to be a legal call? Should we not make it `@Nonnull` and done?

Reply:
Agreed, this parameter must not be null.